src/compiler/checker.ts

```diff
@@ -28898,6 +28898,20 @@ namespace ts {
             });
         }
 
+        function isArgumentsInPropertyInitializerOrClassStaticBlock(node: Node): boolean {

```

Comment:
Why not update `isInPropertyInitializerOrClassStaticBlock`? The logic in there seems questionable, as it `"quit"`s in as soon as it hits a block body that's not part of an arrow function or class static block. I think that meant something like

```ts
declare function condition(): boolean;
declare function doSomething(x: unknown): void;

class C {
    static {
        if (condition()) {
            doSomething(arguments);
        }
    }
}
```

Reply:
No, [it doesn't fail](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzD2C2zwAoBKALngCMccIQpUBuAKFElgRXVPzAcAZRwBbEBgAWWVAHMyADxpoA1qhwB3VNXgA3HFmAd2fTLnyJK8AN7t4D+HAzIY+bgGcP8AMK37joEeGFDYYP6BkY5YiPBkhKjEApQUEVHpDkKiEtKyCrByyBKoGB4UHBmBAL4BUTXV7DVAA). And I find another issue of current implemetation [playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8BQ1XQgFzBglsRKaRGAFjhAHQBGhRqOAZtABQCMAlAXd9KedQG5oAemHoSAewCuIACbQApgCclEpbToBfDdo2ZseaFWgBeaKwFJNQA). Guess I can update `isInPropertyInitializerOrClassStaticBlock` and fix both issues

```ts
class C {
    static {
        this.b
        if (1) {
            this.b; // should error
        }
    }

    static b = 1;
}
```